dts/arm/st/stm32f0.dtsi

```diff
@@ -139,6 +159,7 @@
 		};
 
 		i2c2: i2c@40005800 {
+			i2c-controller;
```

Comment:
i2c-controller is not a simple i2c instance.
Cf https://elixir.bootlin.com/linux/v4.16.12/source/Documentation/devicetree/bindings/i2c/i2c-pxa-pci-ce4100.txt, which is the only available example of use in Linux.
i2c-controller is a controller of i2c bus masters.
So this is not valid here.

Reply:
"This PCI device has three PCI-bars, each bar contains a complete I2C
controller. So we have a total of three"

For me this states: there are three I2C controllers which are the  I2C bus masters of their respective I2C bus. Same as the I2C controller of the STM32 - I2C controller == bus master.

Reply:
To me this is not quite the description of a quite common I2C.
And controller has a specific signification in device tree spefication.
As said previously, not all nodes are thing-controller and we cannot abuse of it.
We need to adapt to existing .dts(i) files (most of them being in Linux today) and not expect
spec to bend to ease code generation in zephyr.
If we need to add one zephyr specific it should be ok. But here you expect most nodes to be modified. And then if every node is a controller, this is the whole -controller concept that need to
be revisited in device tree speficication. I don't think this is a possible path.
If we need more information out of device tree files, we should get help from yaml files.